From 0480147e415efbe4cd85cd445b69f0f1bad05711 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:35 -0400
Subject: [net] sunrpc: Fix a module reference leak in svc_handle_xprt

Message-id: <1403560314-32525-94-git-send-email-bfields@redhat.com>
Patchwork-id: 84549
O-Subject: [Patch RHEL7 093/172] SUNRPC: Fix a module reference leak in svc_handle_xprt
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

If the accept() call fails, we need to put the module reference.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: c789102c20bbbdda6831a273e046715be9d6af79
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/svc_xprt.c b/net/sunrpc/svc_xprt.c
index 614956f..29772e0 100644
--- a/net/sunrpc/svc_xprt.c
+++ b/net/sunrpc/svc_xprt.c
@@ -730,6 +730,8 @@ static int svc_handle_xprt(struct svc_rqst *rqstp, struct svc_xprt *xprt)
   newxpt = xprt->xpt_ops->xpo_accept(xprt);
   if (newxpt)
    svc_add_new_temp_xprt(serv, newxpt);
+  else
+   module_put(xprt->xpt_class->xcl_owner);
  } else if (xprt->xpt_ops->xpo_has_wspace(xprt)) {
   /* XPT_DATA|XPT_DEFERRED case: */
   dprintk("svc: server %p, pool %u, transport %p, inuse=%d\n",
-- 
1.7.1